### ISSUE TITLE
MD032 false positive on sibling list item after an item containing a nested list

### ISSUE THREAD

**Issue by vitaliytv** [rvben/rumdl#898]
## Summary

`MD032` fires on a sibling list item when the **previous sibling item contains a nested list**. Both items belong to the same list, so no blank line is required between them.

`markdownlint` / `markdownlint-cli2` report nothing on the same input.

## Version

- `rumdl 0.2.73` (Homebrew, macOS arm64)
- compared against `markdownlint-cli2` (latest via `npx`)

## Reproduction

`repro.md` (11 lines):

```markdown
- **a** text

  - **n1** nested
  - **n2** nested

  paragraph after nested list

  > blockquote

  tail paragraph
- **b** next sibling
```

```console
$ rumdl check repro.md --disable MD013,MD041
repro.md:11:1: [MD032] List should be preceded by blank line [*]

$ npx markdownlint-cli2 repro.md
repro.md:1 error MD041/first-line-heading/first-line-h1 ...
Summary: 1 issue in 1 file      # MD041 only, no MD032
```

## Expected

No `MD032`. Line 11 (`- **b**`) is the next item of the list opened on line 1 — lines 3-10 are indented continuation content of the first item. Per CommonMark, sibling items of one list do not need a blank line between them.

## Trigger isolation

Removing only the nested list (lines 3-4) makes the same structure clean:

```markdown
- **a** text

  paragraph

  > blockquote

  tail
- **b** next sibling
```

```console
$ rumdl check nonest.md --disable MD013,MD041
Success: No issues found in 1 file
```

So the nested list is the trigger: once it closes, the following sibling seems to be parsed as the start of a new list rather than a continuation of the outer one.

## Note on `rumdl fmt`

`rumdl fmt` resolves this by inserting a blank line before the sibling item. That output is valid and renders identically (HTML from `marked` is byte-identical before and after, since the list is already loose), but the underlying diagnostic still looks incorrect — the original document did not violate MD032.

## Context

Found while migrating a ~377-file documentation tree from `markdownlint-cli2` to `rumdl`. This was the only behavioural divergence on a classic (non-`rumdl`-specific) rule across the entire tree; everything else matched. Happy to test a fix.

**Comment by rvben** [comment]
Thanks @vitaliytv!

Fixed in [952c5079](https://github.com/rvben/rumdl/commit/952c50795102d85b0dc3350eada02983cce94152), released in [v0.2.74](https://github.com/rvben/rumdl/releases/tag/v0.2.74). MD032 now keeps the parent list item open after nested lists, avoiding the false positive on the following sibling item.

Closing as completed.
